nixos/modules/services/web-apps/funkwhale.nix

```diff
@@ -0,0 +1,565 @@
+        '';
+        example = "funkwhale.yourdomain.net";
+      };
+
+      protocol = mkOption {
+        type = types.enum [ "http" "https" ];
+        default = "https";
+        description = ''
+          Web server protocol.
+        '';
+      };
+
+      emailConfig = mkOption {
+        type = types.str;
+        default = "consolemail://";
+        description = ''
+          Configure email sending. By default, it outputs emails to console instead of sending them. See https://docs.funkwhale.audio/configuration.html#email-config for details.
+        '';
+        example = "smtp+ssl://user@:password@youremail.host:465";
+      };
+
+      defaultFromEmail = mkOption {
+        type = types.str;
+        description = ''
+          The email address to use to send system emails.
+        '';
+        example = "funkwhale@yourdomain.net";
+      };
+
+      api = {
+        mediaRoot = mkOption {
+          type = types.str;
+          default = "/srv/funkwhale/data/media";
+          description = ''
+            Where media files (such as album covers or audio tracks) should be stored on your system ? Ensure this directory actually exists.
+          '';
+        };
+
+        staticRoot = mkOption {
+          type = types.str;
+          default = "/srv/funkwhale/data/static";
+          description = ''
+            Where static files (such as API css or icons) should be compiled on your system ? Ensure this directory actually exists.
+          '';
+        };
+
+        djangoSecretKey = mkOption {
+          type = types.str;
+          description = ''
+            Django secret key. Generate one using `openssl rand -base64 45` for example.
+          '';
+          example = "6VhAWVKlqu/dJSdz6TVgEJn/cbbAidwsFvg9ddOwuPRssEs0OtzAhJxLcLVC";
+        };
+      };
+
+      musicDirectoryPath = mkOption {
+        type = types.str;
+        default = "/srv/funkwhale/data/music";
+        description = ''
+          In-place import settings.
+        '';
+      };
+
+      enableRaven = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Sentry/Raven error reporting (server side).
+          Enable Raven if you want to help improve funkwhale by
+          automatically sending error reports to the funkwhale developers Sentry instance.
+          This will help them detect and correct bugs.
+        '';
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.users.funkwhale = {
+      description = "Funkwhale service user";
+      isSystemUser = true;
+      createHome = false;
+      home = cfg.dataDir;
+      group = "funkwhale";
+    };
+
+    users.groups.funkwhale = {};
+
+    services.ntp.enable = true; 
+    services.redis.enable =  true;
+    services.postgresql = {
+      enable = true;
+      authentication = pkgs.lib.mkOverride 10 ''
+        local all all trust
+        host all all ::1/128 trust
```

Comment:
And this here also disables authentication for local connections. This can't stay either.

Reply:
fixed